File: sum_prime.py
#! /usr/bin/python

"""
Sum of Primes

Description:

Write a program to determine the sum of the first 1000 prime numbers.

"""

#Generate prime numbers
def generatePrimes(primes):
	for i in xrange(3,10000, 2):
		if isPrime(i):
			primes.append(i)
		if len(primes) >= 1000:
			break		
			

	
def isPrime(num):
	for i in xrange(3, num/2,2):
		if num % i == 0:
			return False
	return True
	
answer = 0
primeList = [2]
generatePrimes(primeList)

for i in primeList:
	answer += i

print "%i" % answer

File: sum_prime.js
#!/usr/local/bin/node

/*
Sum of Primes

Description:
Write a program to determine the sum of the first 1000 prime numbers
 */

function isPrime(num){
    for(let i = 2; i< num; i++) {
        if (num % i === 0) {
            return false;
        }
    }
    return num > 1;
}

function sumPrimes(){
    let primes = [...Array(10000).keys()].filter(val => isPrime(val));
    primes = primes.slice(0,1000);
    return primes.reduce( (a,b) => { return a+ b;}, 0);
}

console.log(sumPrimes());